### ISSUE TITLE
Enable PipedagConfig construction from dictionary

### ISSUE THREAD

**Issue by windiana42** [pydiverse/pydiverse.pipedag#142]
If someone likes to programmatically produce pipedag configuration, it should be possible to construct a PipedagConfig object from a dictionary. Ideally, it is possible to input PipedagConfig.raw_config into that constructor and yield exactly the same PipedagConfig.config_dict.

An extended version of implementing this feature would offer some programmatic calls which can produce config dictionaries which are good default configurations with very little code.

Hacky workaround code as long as this feature is not available:
```python
    pipedag_config = PipedagConfig()
    pipedag_config.raw_config["table_store_connections"][connection_name]["args"]["url"] = override_engine_url
    pipedag_config.config_dict = pipedag_config._PipedagConfig__parse_config(
        pipedag_config.raw_config
    )
    cfg = pipedag_config.get(instance)
```

**Comment by windiana42** [comment]
Ideally this would allow for much simpler getting started documentation with programmatically created duckdb configuration:

- https://github.com/Quantco/vectorization-tutorial/blob/main/pipedag.yaml
- https://github.com/pydiverse/pydiverse.pipedag/blob/main/docs/source/quickstart.md
- https://github.com/pydiverse/pydiverse.pipedag/blob/main/docs/package/README.md
- https://github.com/pydiverse/pydiverse.pipedag/blob/main/README.md

**Comment by windiana42** [comment]
Closed with #142
